testcases/gcov_setup.py

```diff
@@ -0,0 +1,167 @@
+        k_config_params.extend(["CONFIG_DRM_VIRTIO_GPU=m", "CONFIG_SND_VIRTIO=m", "CONFIG_VIRTIO_VDPA=m", "CONFIG_DRM_QXL=m", "CONFIG_VGA_ARB=m", "CONFIG_VGA_CONSOLE=y"])
+        k_config_params.extend(["CONFIG_INFINIBAND=m", "CONFIG_INFINIBAND_VIRT_DMA=m", "CONFIG_HID=m", "CONFIG_DRM_FBDEV_EMULATION=m", "CONFIG_DRM_KMS_HELPER=m"])
+        k_config_params.extend(["CONFIG_DRM_VKMS=m", "CONFIG_DRM_BOCHS=m", "CONFIG_DRM_CIRRUS_QEMU=m", "CONFIG_FB_MATROX=m", "CONFIG_FB_MATROX_G=m"])
+        k_config_params.extend(["CONFIG_DRM_AST=m"])
+        err_param = []
+        for param in k_config_params:
+            log.info("\n working on param  %s" %param)
+            unset_param = param
+            if param.split("=")[-1] == "y":
+                new_param = param[:-1] + "m"
+            else:
+                new_param = param[:-1] + "y"
+
+            out = int("".join(self.cv_HOST.host_run_command(f"grep -n ^{param} {conf_file} | wc -l")))
+            new_out = int("".join(self.cv_HOST.host_run_command(f"grep -n ^{new_param} {conf_file} | wc -l")))
+
+            if param.split("=")[-1] == "m":
+                param = f"#{param}"
+            if out == 1:
+                if param.split("=")[-1] == "m":
+                    log.info("param found but needs to unset....\n unsetting the param")
+                    self.cv_HOST.host_run_command(f"sed -i 's/^{unset_param}/{param}/g' {conf_file}")
+                else:
+                    log.info("%s parameter already available with same value, So skipping to change it" %param)
+                    continue
+            elif new_out == 1:
+                log.info("%s parameter exists with opposit valuue , changing it" %param)
+                self.cv_HOST.host_run_command(f"sed -i 's/^{new_param}/{param}/g' {conf_file}")
+            else:
+                log.info("%s parameter does not exists, so appending" %param)
+                self.cv_HOST.host_run_command(f"echo {param} >> {conf_file}")
+
+            out_check = int("".join(self.cv_HOST.host_run_command(f"grep -n ^{param} {conf_file} | wc -l")))
+            if out_check == 1:
+                log.info("%s  param added/changed successfully" %param)
+            else:
+                err_param.append(param)
+            log.info("\n\n\n")
+        if err_param:
+            self.fail("few param did not got updated: %s" %err_param)
+
+
+    def build_and_boot(self, config_path):
+        """
+        building and booting the newly installed kernel
+        """
+        self.cv_HOST.host_run_command(f"cd {config_path}")
+        self.cv_HOST.host_run_command("make olddefconfig")
+        cpu_count = self.get_cpu_count()
+        cmd = f"make -j {cpu_count} && make -j modules_install && make install"
+        self.cv_HOST.host_run_command(cmd)
+        self.cv_SYSTEM.goto_state(OpSystemState.OS)
```

Comment:
you all ready in OS state, so this may not reboot system

Reply:
self.cv_SYSTEM.goto_state(OpSystemState.OFF)
   self.cv_SYSTEM.goto_state(OpSystemState.OS)

Reply:
self.cv_SYSTEM.goto_state(OpSystemState.off)
self.cv_SYSTEM.goto_state(OpSystemState.OS)